pkg/common/telemetry/inmem_test.go

```diff
@@ -74,31 +74,44 @@ func TestWarnOnFutureDisable(t *testing.T) {
 
 	// Get a real logrus.Entry
 	logger.SetLevel(logrus.DebugLevel)
-	logger.Debug("boo")
+	logger.Debug("setup log")
 	c := &MetricsConfig{
 		Logger:      hook.LastEntry(),
 		ServiceName: "foo",
 	}
 
 	ir, err := newInmemRunner(c)
 	require.Nil(t, err)
-	assert.Equal(t, 1, len(ir.sinks()))
+	require.Equal(t, 1, len(ir.sinks()))
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
 	go ir.run(ctx)
 
 	// Wait for run to complete set up
-	time.Sleep(100 * time.Millisecond)
+	util.RunWithTimeout(t, 5*time.Second, func() {
```

Comment:
It feels a little gross to me to add an unrelated debug message to the production code path just to facilitate test timing. We should be careful adding log lines that don't have clear debuggability impact. What's wrong with sending the signal more than once while we wait for things to be set up?